GeneralsMD/Code/Libraries/Include/Lib/intrin_compat.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#if !(defined(_MSC_VER) && _MSC_VER < 1300)
+
+#include <cstdint>
+
+#if !defined(_lrotl) && !defined(_WIN32)
+static inline uint32_t _lrotl(uint32_t value, int shift)
+{
+#if defined(__has_builtin) && __has_builtin(__builtin_rotateleft32)
+    return __builtin_rotateleft32(value, shift);
+#else
+    return ((value << shift) | (value >> (32 - shift)));
```

Comment:
Shouldn't this be:
```c++
shift &= 31;
return (value << shift) | (value >> (32 - shift));
```
Otherwise a shift > than 32 could lead to UB.

Reply:
Ok. Added.